src/Explorer/Menus/CommandBar/ConnectionStatusComponent.tsx

```diff
@@ -0,0 +1,90 @@
+import { ProgressIndicator, Spinner, SpinnerSize, Stack, TooltipHost } from "@fluentui/react";
+import * as React from "react";
+import { ConnectionStatusType } from "../../../Common/Constants";
+import { useNotebook } from "../../Notebook/useNotebook";
+import "../CommandBar/ConnectionStatusComponent.less";
+
+export const ConnectionStatus: React.FC = (): JSX.Element => {
```

Comment:
Make this a React class, look at MongoIndexingPolicyComponent for reference maybe. You ll have states and props, changing a state will change UI directly

Reply:
With new version of react we got this latest way [Doing the same].